### ISSUE TITLE
cross-origin error when precaching with google workbox

### ISSUE THREAD

**Issue by germyjen** [cypress-io/cypress#5159]
### Current behavior:

Whenever cypress tries to run tests against a page that uses assets precached with google workbox, the before hook fails, showing this error:
```
 1) Landing Page "before each" hook for "loads the page header":

     Error: Blocked a frame with origin "<url>" from accessing a cross-origin frame.
```

### Desired behavior:

the tests run without throwing the cross-origin error and are able to complete.

### Steps to reproduce: 

1. pre-cache any page element with google workbox using `precache and route`. For example, we've pre-cached our .svg, .css and main.js.map
https://developers.google.com/web/tools/workbox/reference-docs/latest/workbox.precaching#.precacheAndRoute

2. Run a basic test that navigates to the page and verifies a page element. example:
```js
describe('Landing Page', function() {
  beforeEach(function() {
    cy.visit('');
    cy.url().should('eq', `${Cypress.config().baseUrl}/`);
  });

  it('loads the page header', function() {
    cy.getByText('Title').should('exist');
  });
```

### Versions

cypress: 3.4.1
occurs both in electron and chrome
chromewebsecurity: false does not fix the issue

**Comment by azizhk** [comment]
Getting the same error. Am also using Service Workers to precache the content.
```
Error: Blocked a frame with origin "https://test.ssldomain.com:3000" from accessing a cross-origin frame.
 at setWindowDocumentProps (https://test.ssldomain.com:3000/__cypress/runner/cypress_runner.js:78568:42)
 at HTMLIFrameElement.<anonymous> (https://test.ssldomain.com:3000/__cypress/runner/cypress_runner.js:78950:11)
 at HTMLIFrameElement.dispatch (https://test.ssldomain.com:3000/__cypress/runner/cypress_runner.js:21312:27)
 at HTMLIFrameElement.elemData.handle (https://test.ssldomain.com:3000/__cypress/runner/cypress_runner.js:21124:28)
```

This is because we are caching the document (html). Cypress modifies the html and inserts it's own scripts.

I think a temporary fix would be to disable caching html or disable the service worker itself.

**Comment by MartijnHols** [comment]
I ran into this problem during CI since I don't install the Service Worker locally (unless I am actively developing it). Adding `config: chromeWebSecurity=false` to the `cypress-io/github-action@v2` config solved the cross-origin error for me.

After solving the cross-origin error I ran into the issue that Cypress times out waiting for the browser on the next spec. I figured this is probably because Chrome continues to run in the background after being closed when a Service Worker has been installed to be able to handle push notifications (even if your SW doesn't have push notifications). I solved the timeout by unregistering the Service Worker after each spec:

```ts
  after(() => {
    navigator.serviceWorker.getRegistrations().then(function (registrations) {
      registrations.forEach((registration) => registration.unregister())
    })
  })
```
